src/ol/source/stamensource.js

```diff
@@ -88,36 +88,78 @@ ol.source.StamenProviderConfig = {
  */
 ol.source.Stamen = function(options) {
 
-  var i = options.layer.indexOf('-');
-  var provider = i == -1 ? options.layer : options.layer.slice(0, i);
-  goog.asserts.assert(provider in ol.source.StamenProviderConfig,
-      'known provider configured');
-  var providerConfig = ol.source.StamenProviderConfig[provider];
+  /**
+   * url
+   * @type {string}
+   * @private
+   */
 
-  goog.asserts.assert(options.layer in ol.source.StamenLayerConfig,
-      'known layer configured');
-  var layerConfig = ol.source.StamenLayerConfig[options.layer];
+  this.rootUrl_ = goog.isDef(options.url) ? options.url :
+      'https://stamen-tiles-{a-d}.a.ssl.fastly.net/';
```

Comment:
There is a change here in the interpretation of the `url` option.  Previously, if `url` was provided, `layer` would be ignored (or at least not used to configure the URL).  I think it is reasonable to say this is a bug (since `layer` is required, it should not be ignored).

Your change makes it explicit that the `url` option is the "root URL," and I think this change should be documented.

Could you update the documentation in [`olx.js`](https://github.com/SDaron/ol3/blob/stamen-source/externs/olx.js#L4555-L4560)?  The `url` option should now *not* include `{x}`, `{y}` or `{-y}`, and `{z}` placeholders.